packages/cdktf-cli/bin/cmds/ui/terraform-context.tsx

```diff
@@ -514,123 +534,131 @@ export const useTerraform = ({
     }
   };
 
-  const synth = () => {
-    React.useEffect(() => {
-      const invoke = async () => {
-        await execTerraformSynth(false);
-      };
-
-      invoke();
-    }, []);
-
-    return state;
+  return {
+    synth: execTerraformSynth,
+    deploy: execTerraformApply,
+    diff: execTerraformPlan,
+    output: execTerraformOutput,
+    destroy: execTerraformDestroy,
+    init: execTerraformInit,
   };
+};
 
-  const init = () => {
-    React.useEffect(() => {
-      const invoke = async () => {
-        await execTerraformSynth();
-      };
-      invoke();
-    }, []);
+const useRunOnce = <Fn extends (...args: any[]) => any>(
+  fn: Fn,
+  ...args: Parameters<Fn>
+) => {
+  React.useEffect(() => {
+    fn(...args);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []); // only run once on mount – ignore any changes to fn
+};
 
-    React.useEffect(() => {
-      const invoke = async () => {
-        await execTerraformInit();
-      };
+const useRunIf = <Fn extends (...args: any[]) => any>(
```

Comment:
Nitpick: If indicates a conditional for me, but in this case it's mostly temporal, right? So like useRunWhen?

Reply:
Hm, technically it waits for a boolean value which seems more like "if" but in practice it is going to be a temporal thing. Going to adjust it to `When` – I think that suits the context better 👍